docs/src/main/asciidoc/_migration-guide-for-4.0.adoc

```diff
@@ -276,8 +347,48 @@ NOTE: `azure.activedirectory.graph-membership-uri` has been replaced by 2 proper
+[cols="<~,<~", options="header"]
+|===
+|Legacy class |Modern class
+|com.azure.spring.aad.webapi.AADResourceServerWebSecurityConfigurerAdapter | com.azure.spring.cloud.autoconfigure.aad.AadResourceServerWebSecurityConfigurerAdapter
+|com.azure.spring.aad.webapp.AADWebSecurityConfigurerAdapter | com.azure.spring.cloud.autoconfigure.aad.AadWebSecurityConfigurerAdapter
+|com.azure.spring.autoconfigure.aad.AADAppRoleStatelessAuthenticationFilter | com.azure.spring.cloud.autoconfigure.aad.filter.AadAppRoleStatelessAuthenticationFilter
+|com.azure.spring.autoconfigure.aad.AADAuthenticationFilter | com.azure.spring.cloud.autoconfigure.aad.filter.filter.AadAuthenticationFilter
+|com.azure.spring.autoconfigure.aad.AADAuthenticationProperties | com.azure.spring.cloud.autoconfigure.aad.properties.AadAuthenticationProperties
+|com.azure.spring.autoconfigure.aad.UserPrincipal | com.azure.spring.cloud.autoconfigure.aad.filter.UserPrincipal
+|===
+
+==== From azure-spring-boot-starter-active-directory-b2c to spring-cloud-azure-starter-active-directory-b2c
+This guide is intended to assist in the migration to
+link:https://search.maven.org/artifact/com.azure.spring/spring-cloud-azure-starter-active-directory-b2c[spring-cloud-azure-starter-active-directory-b2c] from
+version 3 of link:https://search.maven.org/artifact/com.azure.spring/azure-spring-boot-starter-active-directory-b2c[azure-spring-boot-starter-active-directory-b2c].
+
+* Please refer to <<migration-guide-introduction, introduction>> and <<migration-guide-benefits, migration benefits>> to get a whole picture of the changes in 4.0.
+* Please refer to <<migration-guide-naming, naming changes>> to learn more about the strategy changes in the project naming.
+* Please refer to <<migration-guide-bom, bom>> to learn how to using one BOM for all Spring Cloud Azure libraries.
+* Please refer to <<migration-guide-authentication, authentication>> to learn how to handle authentication in Spring Cloud Azure 4.0.
+* Please refer to <<migration-guide-properties-migration, properties migration>> to learn how to leverage `spring-boot-properties-migrator` during migration.
+* Please refer to <<migration-guide-global-configurations, global configurations>> to learn more about the global and common configuration changes.
+
+[#dependency-spring-cloud-azure-starter-active-directory-b2c]
+===== Dependency changes
+Some unnecessary dependencies were included in the legacy artifacts, which we have removed in the modern Spring Cloud
+Azure 4.0 libraries. Please make sure add the removed dependencies manually to your project to prevent unintentionally
+crash.
+
+.Removed dependencies of spring-cloud-azure-starter-active-directory-b2c
+[cols="<30,<~", options="header"]
+|===
+|Removed dependencies |Description
+|org.springframework.boot:spring-boot-starter-validation |Please include the validation starter if you want to use Hibernate Validator.
```

Comment:
How about we always quotes the "validation starter" if it is not usual work or phrase. idiomatic expression quoted or upper cased looks clear and understandable.